### DISCUSSION TITLE
是否能支持 huggingface/tokenizers

### DISCUSSION THREAD

**Discussion by Liangdi** [zai-org/ChatGLM3#435] [Bad Case]
最近使用 [candle](https://github.com/huggingface/candle) , 在提交 chatglm 需求的时候遇到一个问题，candle 使用 https://github.com/huggingface/tokenizers  这个库， 使用时候需要一个 tokenizer.json , 在 chatglm 中没有这个文件，一些其他模型如：https://huggingface.co/bert-base-chinese ,https://huggingface.co/Salesforce/blip-image-captioning-large 等有相关支持。
看了一下 transformer 文档， 似乎是 fast-tokenziers 这个模块 https://huggingface.co/docs/transformers/fast_tokenizers 

不知道 chatglm 是否能支持这个模块。

candle issue:
https://github.com/huggingface/candle/issues/1177#issuecomment-1789550037

**Comment by Liangdi** [comment]
补充一些内容， transformers 的一些相关代码 https://github.com/huggingface/transformers/blob/main/src/transformers/convert_slow_tokenizer.py

以下是 candle 支持 marian-mt 修改的 convert_slow_tokenizer.py 的代码
https://github.com/huggingface/candle/blob/main/candle-examples/examples/marian-mt/convert_slow_tokenizer.py#L1262C32-L1262C32

**Comment by Liangdi** [comment]
@zRzRzRzRzRzRzR 这个 issue 是受理了还是？

**Comment by zRzRzRzRzRzRzR** [comment]
> @zRzRzRzRzRzRzR 这个 issue 是受理了还

**Comment by zRzRzRzRzRzRzR** [comment]
已经被标注，处理成了关闭，是我这边的问题，已重新打开，明天要做一下适配尝试

**Comment by Liangdi** [comment]
> 已经被标注，处理成了关闭，是我这边的问题，已重新打开，明天要做一下适配尝试

感谢，正巧 Yi-6B 那边已经给了测试的 tokenizer.json 我测试过 ok, 技术上应该走的通  https://github.com/01-ai/Yi/issues/24#issuecomment-1801859811 
要是能把 ChatGLM 使用 candle 部署就方便多了

**Comment by zRzRzRzRzRzRzR** [comment]
收到请求，正在处理，有消息会第一时间回复

**Comment by zRzRzRzRzRzRzR** [comment]
不确定是否支持，可能要改 tokenizer.json。我们参照您最新提供的这份链接让算法同学在看看，回复较慢，望理解

**Comment by Liangdi** [comment]
> 不确定是否支持，可能要改 tokenizer.json。我们参照您最新提供的这份链接让算法同学在看看，回复较慢，望理解

https://huggingface.co/THUDM/chatglm3-6b/discussions/12
这个是你们的新 pr 吗？

**Comment by zRzRzRzRzRzRzR** [comment]
不是，我们看一下

**Comment by zRzRzRzRzRzRzR** [comment]
> > 不确定是否支持，可能要改 tokenizer.json。我们参照您最新提供的这份链接让算法同学在看看，回复较慢，望理解
> 
> https://huggingface.co/THUDM/chatglm3-6b/discussions/12 这个是你们的新 pr 吗？

这个PR可能能够解决你提到的这个问题，算法工程师同事正在审核并验证是否能用，能用会合并

**Comment by Liangdi** [comment]
> > > 不确定是否支持，可能要改 tokenizer.json。我们参照您最新提供的这份链接让算法同学在看看，回复较慢，望理解
> > 
> > 
> > https://huggingface.co/THUDM/chatglm3-6b/discussions/12 这个是你们的新 pr 吗？
> 
> 这个PR可能能够解决你提到的这个问题，算法工程师同事正在审核并验证是否能用，能用会合并

恩，如果你们验证了，直接生成 tokenizer.json 提交在 hf 和 modelscope 呗，这样其他库就可以直接调用了

**Comment by zRzRzRzRzRzRzR** [comment]
> > > > 不确定是否支持，可能要改 tokenizer.json。我们参照您最新提供的这份链接让算法同学在看看，回复较慢，望理解
> > > 
> > > 
> > > https://huggingface.co/THUDM/chatglm3-6b/discussions/12 这个是你们的新 pr 吗？
> > 
> > 
> > 这个PR可能能够解决你提到的这个问题，算法工程师同事正在审核并验证是否能用，能用会合并
> 
> 恩，如果你们验证了，直接生成 tokenizer.json 提交在 hf 和 modelscope 呗，这样其他库就可以直接调用了

对，目前该问题被记录为Bad Case，我们的开发人员正在做这个任务，该问题移交到讨论区

**Reply by Liangdi** [reply]
不知道这块有新的进展么

**Comment by zRzRzRzRzRzRzR** [comment]
这份PR我们没有合并，但是可以自行尝试，由于人手不足，目前应该没有相关计划了

**Comment by ChieloNewctle** [comment]
@Liangdi Hi，[这个 PR](https://huggingface.co/THUDM/chatglm3-6b/discussions/12) 我更新了一下，现在的 fast tokenizer 导出的 `tokenizer.json` 应该可以直接使用了。

**Reply by Liangdi** [reply]
感谢，我去试试
